File: javascript/leetcode/Leetcode_Oct/ArraysHashing/TwoSum.js
// Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

// You may assume that each input would have exactly one solution, and you may not use the same element twice.

// You can return the answer in any order.

const TwoSum = function(nums, target) {
    let arr = [];
    for (let i = 0; i < nums.length; i++) {
        for (let j = i+1; j < nums.length; j++) {
            if (nums[i] + nums[j] === target) {
                arr.push(i, j);
            }
        }
    }
    return arr;
}

console.log(TwoSum([2,7,11,15], 9));

File: Python/leetcode/Roadmap/Arrays_and_Hashing/two_sum.py
# Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

# You may assume that each input would have exactly one solution, and you may not use the same element twice.

# You can return the answer in any order.

class TwoSum:
    def two_sum(self, nums, t):
        for i in range(0, len(nums)-1): 
            for j in range (1, len(nums)):
                if nums[i] + nums[j] == t and i != j:
                    solution = []
                    solution.append(i)
                    solution.append(j)
                    return solution


two = TwoSum()
print(two.two_sum([3,3],6))